### DISCUSSION TITLE
how set nlohmann json read with case-insensitive?

### DISCUSSION THREAD

**Discussion by zhangxiaobin2** [nlohmann/json#3034] [Q&A] [answered]
Some times, we need to read field from json file with case-insensitive?
like:
{
  "Name" : "xxxxxxx"
}
We want to jsonObj["name"]  or jsonObj["Name"] to read the value of "Name" with case-insensitive. Is there any way to realize within nlohmann json?

Thanks :)

**Comment by nlohmann** [comment, marked as answer]
I see two ways for this:

- Add wrapper functions around the object access that take care that keys are only stored in lower case. You'd then always call these functions instead of the native `operator[]`, `at`, or `value`.
- Define a version of `std::map` that takes care of this. You can then pass this type as template parameter to `basic_json` to define your own "case-insensitive JSON" version.

**Reply by zhangxiaobin2** [reply]
thank you very much
